libraries/SPI/SPI.cpp

```diff
@@ -74,13 +75,17 @@ void SPIClass::setHwCs(bool use) {
 }
 
 void SPIClass::beginTransaction(SPISettings settings) {
-    while(SPI1CMD & SPIBUSY) {}
+    while(inTransaction || SPI1CMD & SPIBUSY) {
+        yield();
```

Comment:
> the SPI itself is interrupt safe

unfortunately it is not when software CS and/or multiple devices are used.
Consider interrupt is coming while inside `loop` there is a write (of several lines one by one) to LCD and it is right in the middle of that block (properly protected with `beginTransaction` `endTransaction`):
 * The CS for the LCD is held down and write is started
 * The interrupt routine is called and it successfully starts new transaction after the write is complete
 * The interrupt routine is taking the sensor CS low, but the LCD CS is still low
 * writing garbage to LCD or worse - damaging MISO of the devices if they both try send data
 
The only other way (I can think of) to solve the problem is to disable the interrupts during the SPI transaction. Should I make that change instead?